server/middlewares/validators/videos/videos.ts

```diff
@@ -126,6 +129,92 @@ const videosAddValidator = getCommonVideoEditAttributes().concat([
   }
 ])
 
+const videosAddResumableValidator = getCommonVideoEditAttributes().concat([
+  body('name')
+    .trim()
+    .custom(isVideoNameValid)
+    .withMessage('Should have a valid name'),
+  body('channelId')
+    .customSanitizer(toIntOrNull)
+    .custom(isIdValid).withMessage('Should have correct video channel id'),
+
+  async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+    const videoFileMetadata = {
+      mimetype: req.body.mimeType,
+      size: req.body.size,
+      originalname: req.body.name
+    } as express.FileUploadMetadata
+    const user = res.locals.oauth.token.User
+    const file: Express.Multer.File & { id: string, metadata: any } = req.body
+    file.path = `${CONFIG.STORAGE.VIDEOS_DIR}${file.id}`
+
+    if (
+      !file.metadata.isPreviewForAudio &&
+      !await doesVideoChannelOfAccountExist(file.metadata.channelId, user, res)
+    ) return
+
+    if (!await isVideoAccepted(req, res, file)) {
+      await clearUploadFile(file.path)
+      return res.status(HttpStatusCode.BAD_REQUEST_400).json({ error: 'Video was refused.' })
+    }
+
+    if (file.metadata.isPreviewForAudio) {
+      const filename = `${file.id}-${uuidv4()}`
+      file.filename = filename
+      try {
```

Comment:
I don't think you should catch the error, the middleware should handle the error to automatically returns 500 and log the error